DNN Platform/Modules/Journal/Components/Utilities.cs

```diff
@@ -1,355 +1,374 @@
+        {
+            url = url.Trim();
+            if (!url.StartsWith("http"))
+            {
+                url = "http://" + url;
+            }
+
+            if (url.Contains("https://"))
+            {
+                url = url.Replace("https://", "http://");
+            }
+
+            if (url.Contains("http://http://"))
+            {
+                url = url.Replace("http://http://", "http://");
+            }
+
+            if (!(url.IndexOf("http://") == 0))
+            {
+                url = "http://" + url;
+            }
+
+            Uri objURI = null;
+
+            objURI = new Uri(url);
+
+            return url;
+        }
+
+        internal static LinkInfo GetLinkData(string URL)
+        {
+            string sPage = GetPageFromURL(ref URL, string.Empty, string.Empty);
+            LinkInfo link = new LinkInfo();
+            if (string.IsNullOrEmpty(sPage))
+            {
+                return link;
+            }
+
+            string sTitle = string.Empty;
+            string sDescription = string.Empty;
+            string sImage = string.Empty;
+
+            if (URL.Contains("http://"))
+            {
+                URL = URL.Replace("http://", "//");
+            }
+
+            if (URL.Contains("https://"))
+            {
+                URL = URL.Replace("https://", "//");
+            }
+
+            link.URL = URL;
+            link.Images = new List<ImageInfo>();
+            Match m = PageRegex.Match(sPage);
+            if (m.Success)
+            {
+                link.Title = m.Groups[2].ToString().Trim();
+            }
+
+            MatchCollection matches = default(MatchCollection);
+            matches = MetaRegex.Matches(sPage);
+            int i = 0;
+            foreach (Match match in matches)
+            {
+                string sTempDesc = match.Groups[0].Value;
+                foreach (Match subM in MetaSubRegex.Matches(sTempDesc))
+                {
+                    if (subM.Groups[4].Value.Equals("OG:DESCRIPTION", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        link.Description = subM.Groups[9].Value;
+                    }
+                    else if (subM.Groups[4].Value.Equals("DESCRIPTION", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        link.Description = subM.Groups[9].Value;
+                    }
+
+                    if (subM.Groups[4].Value.Equals("OG:TITLE", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        link.Title = subM.Groups[9].Value;
+                    }
+
+                    if (subM.Groups[4].Value.Equals("OG:IMAGE", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        sImage = subM.Groups[9].Value;
+                        ImageInfo img = new ImageInfo();
+                        if (sImage.Contains("http://"))

```

Comment:
If the image path has a set protocol, I really don't think we should be changing it.  If no protocol, fine add the // to assume site protocol, but blindly changing to match site protocol isn't a great idea

Reply:
> @thabaum Close, the suggestion is to change them all to
> 
> `//domain.com/pic.jpg`
> 
> Defaulting everything to https breaks ALL sites that don't have HTTPS, defaulting to HTTP breaks SOME sites with SSL. Using the // syntax allows the protocol of the page being loaded to dictate https or HTTP

@mitchelsellers 

I figured the website serving the image would set this upon request.  If the requested web server is enabled to serve HTTPS it would serve it that way or serve it HTTP.  DNN would just ask for '//' and allow the servers to figure out which is best upon request.

so your saying only the www.domainrequest.com images should contain the '//' however I understood 'ALL' should have '//' which would then allow any images requested to come as secured or insecured depending on the page requesting the image.  What I am trying to resolve is unsecured images warning on pages that allow only secured content.

I don't believe allowing images in 'http://' would fix this issue.   The problem is unsecured content on pages (images).  At the time the url is "prepared" it saves this url to the database.  Right now it ONLY saves 'http://' which is way off from being a secured website approach.

What do you recommend?

Reply:
if else statements for each variation of a url link to prepare for?  As in a url without 'http' just add '//' and rules that contain 'http' save with the http and urls with 'https' save with https I suppose maybe the solution.  But for my sites I really want only HTTPS images served or no images to avoid the page looking insecured.

So maybe a switch in the panel to dictate the behavior of how these image urls are saved.  Default like it is or allow server protocol to decide.  I just see the current version as not being secured at all serving content via HTTP and so it makes a site page look scary flashing warnings in browsers.